paasta_tools/cli/cmds/remote_run.py

```diff
@@ -0,0 +1,318 @@
+from paasta_tools.utils import load_system_paasta_config
+from paasta_tools.utils import paasta_print
+from paasta_tools.utils import PaastaColors
+from paasta_tools.utils import PaastaNotConfiguredError
+from paasta_tools.utils import SystemPaastaConfig
+from paasta_tools.utils import DEFAULT_SOA_DIR
+from paasta_tools.utils import validate_service_instance
+
+from paasta_tools.native_mesos_scheduler import create_driver_with
+from paasta_tools.native_mesos_scheduler import compose_job_id
+from paasta_tools.native_mesos_scheduler import load_paasta_native_job_config
+
+
+class PaastaAdhocScheduler(Scheduler):
+
+    def __init__ (self, command, service_config, system_paasta_config, dry_run):
+        self.command = command
+        self.service_config = service_config
+        self.system_paasta_config = system_paasta_config
+        self.dry_run = dry_run
+        self.launched = False
+        self.status = None
+
+
+    def registered(self, driver, framework_id, master_info):
+        paasta_print("Registered with framework id: {}".format(framework_id))
+
+
+    def resourceOffers(self, driver, offers):
+        if self.launched:
+            return
+
+        paasta_print("Recieved resource offers: {}".format([o.id.value for o in offers]))
+        offer = offers[0]
+        task = self.new_task(offer)
+        paasta_print("Launching task {task} "
+                     "using offer {offer}.".format(task=task.task_id.value,
+                                                   offer=offer.id.value))
+
+        self.launched = True
+        if self.dry_run:
+            paasta_print("Not doing anything in dry-run mode."
+                         "Would have launched: {task}".format(task=task))
+            self.status = 0
+            driver.stop()
+        else:
+            driver.launchTasks(offer.id, [task])
+
+
+    def statusUpdate(self, driver, update):
+        paasta_print("Mesos Scheduler: task %s is in state %d" % (update.task_id.value, update.state))
+
+        if update.state == mesos_pb2.TASK_FINISHED:
+            self.status = 0
+        elif update.state == mesos_pb2.TASK_FAILED or update.state == mesos_pb2.TASK_ERROR:
+            paasta_print("Task failed: %s" % update.message)
+            self.status = 1
+        elif update.state == mesos_pb2.TASK_LOST or update.state == mesos_pb2.TASK_KILLED:
+            self.status = 1
+            # self.launched = False
+
+        if self.status != None:
+            paasta_print("Task %s is finished." % update.task_id.value)
+            # for stream in mesos.cli.cluster.files(flist=['stdout','stderr'], fltr=update.task_id.value):
+            #     print("Printing %s for task %s" % (stream[0].path, update.task_id.value))
+            #     for line in stream[0].readlines():
+            #         print line
+            driver.stop()
+
+
+    def new_task(self, offer):
+        task = self.service_config.base_task(self.system_paasta_config, portMappings=False)
```

Comment:
it would be nice if this wasn't bound to a class, but was just a function called ``build_base_task`` that took a ``system_paasta_config`` and a ``service_config`` and returned a new Mesos Task. That way you wouldn't have to depend on this class in the from the native scheduler code. Maybe a problem for another day, but there is definitely some refactoring to be done to make the building of new frameworks easier